samples/Shared/UnitTestFramework/GenericAnalyzerTest.cs

```diff
@@ -0,0 +1,1355 @@
+        /// This collection is only used when <see cref="BatchFixedSources"/> differs from
+        /// <see cref="FixedSources"/>.</para>
+        /// </remarks>
+        public List<DiagnosticResult> BatchRemainingDiagnostics { get; } = new List<DiagnosticResult>();
+
+        /// <summary>
+        /// Gets or sets a value indicating whether exclusions for generated code should be tested automatically. The
+        /// default value is <see langword="true"/>.
+        /// </summary>
+        public bool VerifyExclusions { get; set; } = true;
+
+        /// <summary>
+        /// Gets a collection of diagnostics to explicitly disable in the <see cref="CompilationOptions"/> for projects.
+        /// </summary>
+        public List<string> DisabledDiagnostics { get; } = new List<string>();
+
+        /// <summary>
+        /// Gets or sets the index of the code fix to apply.
+        /// </summary>
+        public int? CodeFixIndex { get; set; }
+
+        /// <summary>
+        /// Sets the expected output source file for code fix testing.
+        /// </summary>
+        /// <seealso cref="FixedSources"/>
+        public string FixedCode
+        {
+            set
+            {
+                Assert.Empty(FixedSources);
+                if (value != null)
+                {
+                    FixedSources.Add(value);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets the set of expected output files for code fix testing. Files may be added to this list using one of the
+        /// <see cref="SourceFileList.Add"/> methods.
+        /// </summary>
+        public SourceFileList FixedSources { get; }
+
+        /// <summary>
+        /// Sets the expected output source file after a Fix All operation is applied.
+        /// </summary>
+        /// <seealso cref="BatchFixedSources"/>
+        public string BatchFixedCode
+        {
+            set
+            {
+                Assert.Empty(BatchFixedSources);
+                if (value != null)
+                {
+                    BatchFixedSources.Add(value);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets the set of expected output files after a Fix All operation is applied. Files may be added to this list
+        /// using one of the <see cref="SourceFileList.Add"/> methods.
+        /// </summary>
+        /// <remarks>
+        /// <para>By default, Fix All operations are expected to produce the same result as incremental fix operations.
+        /// If this collection is not specified for the test, <see cref="FixedSources"/> provides the expected test
+        /// results for both incremental and Fix All scenarios.</para>
+        /// </remarks>
+        public SourceFileList BatchFixedSources { get; }
+
+        /// <summary>
+        /// Gets or sets the number of code fix iterations expected during code fix testing.
+        /// </summary>
+        /// <remarks>
```

Comment:
@dotnet/roslyn-ide @kuhlenh This documentation will never be seen by users because it's in a `<remarks>` section.